userspace/libsinsp/test/filter_ppm_codes.ut.cpp

```diff
@@ -18,193 +18,263 @@ limitations under the License.
+    const set_t all_set = libsinsp::events::all_event_set();
+    const set_t close_set = { PPME_SYSCALL_CLOSE_E, PPME_SYSCALL_CLOSE_X };
+    const set_t openat_set = {
         PPME_SYSCALL_OPENAT_E, PPME_SYSCALL_OPENAT_X,
         PPME_SYSCALL_OPENAT_2_E, PPME_SYSCALL_OPENAT_2_X };
+    virtual set_t filter_set(const std::string filter) const
+    {
+        return libsinsp::filter::ast::ppm_event_codes(
+            libsinsp::filter::parser(filter).parse().get());
+    }
+};
+
+// helps testing that ppm_event_codes can be obtained by searching for
+// ppm_sc_codes in a filter, and then using the
+// libsinsp::events::sc_set_to_event_set conversion utility
+struct testdata_event_set_converted: testdata_event_set
+{
+    virtual set_t filter_set(const std::string filter) const override
+    {
+        testdata_sc_set s;
+        return libsinsp::events::sc_set_to_event_set(s.filter_set(filter));
+    }
+};
+
+// helps testing that ppm_sc_codes can be obtained by searching for
+// ppm_event_codes in a filter, and then using the
+// libsinsp::events::event_set_to_sc_set conversion utility
+struct testdata_sc_set_converted: testdata_sc_set
+{
+    virtual set_t filter_set(const std::string filter) const override
+    {
+        testdata_event_set s;
+        return libsinsp::events::event_set_to_sc_set(s.filter_set(filter));
+    }
+};
+
+// helpers to make comparisons easier and more expressive in case of fail
+#define ASSERT_FILTER_EQ(t, a, b) { ASSERT_EQ(t.filter_set(a), t.filter_set(b)); }
+#define ASSERT_FILTER_SET_EQ(t, a, b) { ASSERT_EQ(t.filter_set(a), b); }
 
-    auto not_openat = libsinsp::events::all_event_set().diff(openat_only);
+// helper for making sure tests are run on multiple testdata,
+// so that we're sure that the tests covers the high-level semantics no
+// matter how events are represented. At the same time, this is supposed to
+// stress the ppm_sc_code <-> ppm_event_code conversions.
+//
+// NOTE: tests relying on the "_converted" testdata play with converting sets of
+// ppm_sc_code <-> ppm_event_code, which in general can cause information loss.
+// However, in none of the tests below this is significant, because no filter
+// deals with corner cases such as generic events, meta events, etc.
+// However, the current implementation would still make "_converted" tests fail
+// due to the conversion logic still having some faults.
+// Fixes are in the working in https://github.com/falcosecurity/libs/pull/889.
+// todo(jasondellaluce): enabled "_converted" tests once pull#889 gets merged
+#define TEST_CODES(test_suite_name, test_name) \
+    template <typename T> void test_##test_name(); \
+    TEST(test_suite_name, sc_##test_name) {test_##test_name<testdata_sc_set>();}; \
+    TEST(test_suite_name, event_##test_name) {test_##test_name<testdata_event_set>();}; \
+    /* TEST(test_suite_name, sc_converted_##test_name) {test_##test_name<testdata_sc_set_converted>();}; */ \
```

Comment:
These two tests must be enabled once we merge https://github.com/falcosecurity/libs/pull/889.
Until then, the conversion will have information loss for old event versions, however I didn't feel like blocking this PR due to these failing tests, as the effort is not correlated to the one of #889.